examples/minimq-stm32h7/src/main.rs

```diff
@@ -213,19 +202,27 @@ const APP: () = {
             }
 
             if tick && (time % 1000) == 0 {
+                if client.is_connected().unwrap() {
+                    info!("connected!");
+                }
+                else{
+                    info!("not connected!");
+
+                }
                 client
                     .publish("nucleo", "Hello, World!".as_bytes(), QoS::AtMostOnce, &[])
                     .unwrap();
 
-                let temperature = Temperature {
-                    temperature_c: c.resources.si7021.temperature_celsius().unwrap(),
+                let random = Random {
+                    random: c.resources.rng.gen().ok()
                 };
-                let temperature: String<consts::U256> =
-                    serde_json_core::to_string(&temperature).unwrap();
+
+                let random: String<consts::U256> =
+                    serde_json_core::to_string(&random).unwrap();
                 client
                     .publish(
-                        "temperature",
-                        &temperature.into_bytes(),
+                        "random",
+                        &random.into_bytes(),
                         QoS::AtMostOnce,
                         &[],
                     )
```

Comment:
Can you add a handle of `client.poll()` for `Error::Disconnected` as well? That shouldn't cause a panic - it's normal behavior when the broker disconnects.